source/frontend/src/features/properties/map/project/update/UpdateProjectContainer.tsx

```diff
@@ -1,74 +1,93 @@
 
 export interface IUpdateProjectContainerProps {
-  initialValues: ProjectForm;
-  projectStatusOptions: SelectOption[];
-  projectRegionOptions: SelectOption[];
-  validationSchema?: any | (() => any);
-  onSubmit: (values: ProjectForm, formikHelpers: FormikHelpers<ProjectForm>) => void | Promise<any>;
+  project: Api_Project;
+  onSuccess: () => void;
 }
 
-const UpdateProjectContainer = React.forwardRef<FormikProps<any>, IUpdateProjectContainerProps>(
-  (props, formikRef) => {
-    const {
-      initialValues,
-      projectStatusOptions,
-      projectRegionOptions,
-      validationSchema,
-      onSubmit,
-    } = props;
+const UpdateProjectContainer = React.forwardRef<
+  FormikProps<ProjectForm>,
+  IUpdateProjectContainerProps
+>((props, formikRef) => {
+  const { project, onSuccess } = props;
+  const {
+    updateProject: { execute: updateProject },
+  } = useProjectProvider();
 
-    const handleSubmit = (values: ProjectForm, formikHelpers: FormikHelpers<ProjectForm>) => {
-      onSubmit(values, formikHelpers);
-    };
+  const { getOptionsByType, getByType } = useLookupCodeHelpers();
 
-    return (
-      <Formik<ProjectForm>
-        enableReinitialize
-        innerRef={formikRef}
-        initialValues={initialValues}
-        validationSchema={validationSchema}
-        onSubmit={handleSubmit}
-      >
-        {formikProps => (
-          <>
-            <Container>
-              <Section>
-                <SectionField label="Project name" required={true}>
-                  <Input field="projectName" />
-                </SectionField>
-                <SectionField label="Project number">
-                  <Input field="projectNumber" placeholder="if known" />
-                </SectionField>
-                <SectionField label="Status" required={true}>
-                  <Select
-                    field="projectStatusType"
-                    options={projectStatusOptions}
-                    placeholder="Select..."
-                  />
-                </SectionField>
-                <SectionField label="MoTI region" required={true}>
-                  <Select field="region" options={projectRegionOptions} placeholder="Select..." />
-                </SectionField>
-                <SectionField label="Project summary">
-                  <MediumTextArea field="summary" />
-                </SectionField>
-              </Section>
-            </Container>
-          </>
-        )}
-      </Formik>
-    );
-  },
-);
+  const intialValues = ProjectForm.fromApi(project);
+  const projectStatusTypeCodes = getOptionsByType(API.PROJECT_STATUS_TYPES);
+  const regionTypeCodes = getByType(API.REGION_TYPES).map(c => mapLookupCode(c));
+
+  const handleSubmit = async (values: ProjectForm, formikHelpers: FormikHelpers<ProjectForm>) => {
+    try {
+      const updatedProject = values.toApi();
+      const response = await updateProject(updatedProject);
+
+      if (!!response?.id) {
+        formikHelpers?.resetForm();
```

Comment:
hmm, you may want to reset the form the value returned by the response, otherwise the form may flicker during the transition from edit back to view mode.

Reply:
Project in context get's updated after the update, which is later passed to the Form.